module.go

```diff
@@ -391,3 +391,53 @@ func (m *Module) Iterator() *BPFObjectIterator {
 		prevMap:  nil,
 	}
 }
+
+func (m *Module) linkExist(prog *BPFProg) bool {
+	for _, link := range m.links {
+		if link.GetProg() == prog {
```

Comment:
I suspect that this can be misleading or break if the libbpf changes. What guarantees the same pointer coming from it? What do you think of checking against the prog name?

Reply:
> I suspect that this can be misleading or break if the libbpf changes. What guarantees the same pointer coming from it? What do you think of checking against the prog name?
> 
> --- EDIT
> 
> I meant this deal with prog returned by NextProgram(), so it wraps again a new BPFProg with a pointer provided from the libbpf. Just being extra cautious here.

you mean like?:
if link.GetProgName() == prog.Name()

Reply:
if possible link.prog.Name(), without another helper.

Reply:
> if possible link.prog.Name(), without another helper.
Got it!

Reply:
> > if possible link.prog.Name(), without another helper.
> > Got it!
done!